### ISSUE TITLE
Issue related to free text search between 1.5.1 and 1.7.4

### ISSUE THREAD

**Issue by bbimber** [GMOD/jbrowse-components#2933]
We use free-text search in our JBrowse instance, including indexing of genes from GFF files. This worked well in our embedded instance using JBrowse 1.5.1 (core, lineargenomeview, and cli). We are trying to upgrade to 1.7.4 and something is broken in this feature. Here are some details:

- The primary symptom is that when one types into the RefNameAutocomplete box, while contigs will appear, no other search terms work. 
- I first debugged the trix indexes created from each version. They are identical, except for one space. We are trying to index a dummy automated test GFF file (attached). When processed with @jbrowse/cli 1.5.1 vs. 1.7.4 the trix files are virtually identical. However, the ixx file from 1.7.4 contains one additional space. Here is 1.5.1:
```
gag  0000000000
```
vs 1.7.4:
```
gag   0000000000
```
I'm not sure if github will replicate this, but in the second, there's 3 spaces after 'gag', vs. two. While I dont know if one expects this change or not, this does not appear to explain the problem. if I manually edit this file, the client-side problem remains.

FWIW, here is the ix file, which has the feature 'gag':
```
gag ["SIVmac239_Test%3A10373..10493"|"4F42A8EE-A537-103A-812A-FEB3ACE7E048fakeData.gff.gz"|"Gag"],1 ["SIVmac239_Test%3A2691..2805"|"4F42A8EE-A537-103A-812A-FEB3ACE7E048fakeData.gff.gz"|"Gag"],2 ["SIVmac239_Test%3A4165..4237"|"4F42A8EE-A537-103A-812A-FEB3ACE7E048fakeData.gff.gz"|"Gag"],3 ["SIVmac239_Test%3A5251..5370"|"4F42A8EE-A537-103A-812A-FEB3ACE7E048fakeData.gff.gz"|"Gag"],4 ["SIVmac239_Test%3A8510..8630"|"4F42A8EE-A537-103A-812A-FEB3ACE7E048fakeData.gff.gz"|"Gag"],5
```

Can you suggest any debugging steps I should check on the client to explain why the indexing isnt getting picked up on the client? Thanks in advance for any ideas.

[4F42A8EE-A537-103A-812A-FEB3ACE7E048fakeData.gff.gz](https://github.com/GMOD/jbrowse-components/files/8548720/4F42A8EE-A537-103A-812A-FEB3ACE7E048fakeData.gff.gz)

**Comment by cmdcolin** [comment]
there is probably be a bug with the new prefix size option...can try --prefixSize 5 to use the old default

**Comment by bbimber** [comment]
OK, but how would this matter for the client?

**Comment by cmdcolin** [comment]
The client hardcode's the prefix size here https://github.com/GMOD/trix-js/blob/main/src/index.ts#L3

I believe it can be auto calculated though (lineLength-8 for the 8 chars of the hex address in the ixx)

**Comment by cmdcolin** [comment]
To be clear the text-index updated it to 6 and it was a oversight that it was hard coded elsewhere!

**Comment by cmdcolin** [comment]
if you run `yarn upgrade` on your codebase you can get an upgraded version of @gmod/trix v2.0.3 that fixes this now:)

**Comment by bbimber** [comment]
@cmdcolin Thanks for the quick update; however, neither using --prefixSize 5 not upgrading to trix 2.0.3 changes the client-side behavior. When I look in the network window, I think it's downloading the correct ixx files. In this example, it has the iix with "gag  0000000000". However, when I type into the search box it doesnt report any hits. Is there any client-side code I should check out or things I could try debugging?

**Comment by cmdcolin** [comment]
if you look at the developer console in chrome, and see if there are error messages it might help. in my testing, it was producing errors that indicated HTTP 416 Range not satisfiable errors for some queries

if you are getting those, it would probably indicate that the code is not updated. if you are not getting those, then it might be helpful to see example datasets or reproducition?


here example of digging into the stack trace for with the un-updated @gmod/trix package.

screenshot showing the Range not satisfiable error with stack trace
![Screenshot from 2022-04-25 10-45-40](https://user-images.githubusercontent.com/6511937/165135636-56bd52f3-92e7-4a34-82d2-c8a994b09792.png)

screenshot showing the debugger after clicking into trix-js code, showing the un-updated prefix snippet
![Screenshot from 2022-04-25 10-46-03](https://user-images.githubusercontent.com/6511937/165135651-5ab1ff56-fbf1-4a64-bbee-7fcddb1e3ff6.png)

**Comment by bbimber** [comment]
there is no overt error being reported or logged on the client. i'll try to find time to do more debugging this afternoon.

**Comment by cmdcolin** [comment]
I think I might have found the issue. I tested out the gff that you provided and found a probably fix here https://github.com/GMOD/trix-js/pull/15

Will check over and release a new update

**Comment by bbimber** [comment]
ok. is this a deliberate change relative to earlier versions?

**Comment by cmdcolin** [comment]
if you mean why the bug was introduced, it was an unintentional regression in trix-js 2.0.0. @jbrowse/react-linear-genome-view@v1.5.1 used trix-js v1.0.0 which probably didn't have this particular bug. 2.0.0 was needed because there are bugs in 1.0.0 that resulted in infinite loops (xref https://github.com/GMOD/jbrowse-components/issues/2592)

**Comment by cmdcolin** [comment]
fix published now as @gmod/trix@2.0.4 :)

**Comment by bbimber** [comment]
@cmdcolin Thanks! I havent been able to run this locally, but our automated tests just passed, so this is a good sign.
